config/config_test.go

```diff
@@ -538,7 +540,9 @@ func TestLoadConfig(t *testing.T) {
 
 	// String method must not reveal authentication credentials.
 	s := c.String()
-	if strings.Contains(s, "admin_password") {
+	secretRe := regexp.MustCompile("<secret>")
+	matches := secretRe.FindAllStringIndex(s, -1)
+	if len(matches) < 5 {
 		t.Fatalf("config's String method reveals authentication credentials.")
```

Comment:
You might also want to test that the original passwords aren't present. That'll be a bit more resilient to someone adding new fields and not updating the number.